### ISSUE TITLE
If update project setting, VC++ Directories will reset...

### ISSUE THREAD

**Issue by KavinHan** [openframeworks/projectGenerator#88]
**Before update project setting use project generator you can seeinclude and library directories has cuda path**
<img width="614" alt="1" src="https://cloud.githubusercontent.com/assets/3361694/11034281/b93f7ec4-872d-11e5-8a03-4dbb45b05ab1.png">

**If finish update project setting use project generator.(add one addons)
you can see the cuda path is gone...**
![image](https://cloud.githubusercontent.com/assets/3361694/11034270/adee49a6-872d-11e5-9d1a-932bcc22f531.png)

----
SO: finally the result is => 
If manually add some path to project setting, then will be reset when change project setting use project generator.

**Comment by arturoc** [comment]
the visual studio plugin allows to add and remove addons and will not touch any other settings, check the setup guide for visual studio to see how to install it and use it

**Comment by KavinHan** [comment]
Thank you @arturoc ^^.

The vs plugin works fine.
But if i add addons use vs plugin, then will add absolute path to project setting.
like this:
![image](https://cloud.githubusercontent.com/assets/3361694/11034551/b87de4d8-872f-11e5-9e8c-a27155241db9.png)

I want add relative path.
like this:
```
..\..\..\addons\ofxXmlSetting\src
```
=,=

So before i use project generator add addons...

**Comment by KavinHan** [comment]
@arturoc I get a problem about vs plugin, but I don't know where to submit it.

this link https://github.com/openframeworks/vsplugin not work...

The problem is:

If i create project like this:(change workspace myApps to new one)
<img width="707" alt="2" src="https://cloud.githubusercontent.com/assets/3361694/11034708/f8f37266-8730-11e5-8d1e-8ff77db1eb40.png">

Will show this error:
<img width="494" alt="3" src="https://cloud.githubusercontent.com/assets/3361694/11034713/02387bd2-8731-11e5-84be-992f27a6e4b9.png">

And the directory structure like this:
<img width="107" alt="4" src="https://cloud.githubusercontent.com/assets/3361694/11034728/1e1c05da-8731-11e5-8848-a91805db0e10.png">
 
You can see there has same name directory.
```
G:\of_v0.9.0_vs_release\apps\NewWorkSpace\TryChangeWorkSpace\TryChangeWorkSpace
```

**Comment by ofZach** [comment]
about the PG, not the VS plugin, we are taking a template and adding any addons you want to it, and when you call update, we are basically recreating the project so those project settings are getting lost.  You can always add flags to a given addon using the addon config file if you want those settings to be passed on to the project file.   In your case, maybe you could make an addon which represents what you want the project to have in terms of include flags, linking, etc and then add it like any other addon.

in other words, the PG is not really parsing "your" project file on update, it's just examining the "src" folder, and adding any addons you have to the project to make a new project which is why changes get over-ridden.  There might be down the line a way to describe per project additions similar to what the addon can define in terms of includes, etc that could help with this kind of situation.

**Comment by arturoc** [comment]
you have to uncheck the `create directory for solution` option in the first screen when setting the name and path. i think it's documented in the visual studio guide

i'll upload the source for the plugin later today to it's own repo

**Comment by ofZach** [comment]
(one thing I'd like to investigate is "config.make" which be useful to allow you to set per project include or linking paths that would get reflected in this final project file... I haven't really spent time with this vs the addon addon_config.mk)

**Comment by KavinHan** [comment]
@arturoc If i uncheck it's works good. My mistake.. 
Thank u again..^^

**Comment by KavinHan** [comment]
@ofZach: Thanks for u detailed explanation.^^
I will try to change the code to addons~ :)
